### PR TITLE
Re-add keybindings to new tab dropdown

### LINKED ISSUE #3603: Dropdown menu no longer displays bound keys (after #3391)
This no longer works on 0.6.3181.0; possible regression from #3391?

image

If I write a new binding with the legacy name `newTabProfile3`, I get this:

image

### PR DESCRIPTION
**Pull request by zadjii-msft**
## Summary of the Pull Request

With #3391, I almost certainly regressed the ability for the new tab dropdown to display the keybindings for each profile. This adds them back.

## PR Checklist
* [x] Closes #3603
* [x] I work here
* [ ] Tests added/passed
* [n/a] Requires documentation to be updated

## Detailed Description of the Pull Request / Additional comments

Now, we can lookup keybindings not only for `ShortcutAction`s, but also `ActionAndArgs`s, so we can look up the binding for an action with a particular set of arguments.

### CODE REVIEW

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 1]
```diff
@@ -253,7 +254,21 @@ namespace winrt::TerminalApp::implementation
...
-                auto profileKeyChord = keyBindings.GetKeyBinding(action);
+                // First, attepmt to search for the keybinding for the simple
+                // NewTabProfile0-9 ShortcutActions.
+                auto profileKeyChord = keyBindings.GetKeyBindingForAction(action);
+                if (!profileKeyChord)
+                {
+                    // If NewTabProfileN didn't have a binding, look for a
+                    // keychord that is bound to the equivalent
+                    // NewTab(ProfileIndex=N) action
+                    auto actionAndArgs = winrt::make_self<winrt::TerminalApp::implementation::ActionAndArgs>();
```
Comment (DHowett-MSFT) [inline src/cascadia/TerminalApp/TerminalPage.cpp]: omg the expense

File: src/cascadia/TerminalApp/AppKeyBindings.cpp [hunk 2]
```diff
@@ -40,6 +41,27 @@ namespace winrt::TerminalApp::implementation
         }
         return { nullptr };
     }
+    // Method Description:
```
Comment (carlos-zamora) [inline src/cascadia/TerminalApp/AppKeyBindings.cpp]: nit: add a newline before this

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 3]
```diff
@@ -253,7 +254,21 @@ namespace winrt::TerminalApp::implementation
             {
                 // enum value for ShortcutAction::NewTabProfileX; 0==NewTabProfile0
                 const auto action = static_cast<ShortcutAction>(profileIndex + static_cast<int>(ShortcutAction::NewTabProfile0));
-                auto profileKeyChord = keyBindings.GetKeyBinding(action);
+                // First, attepmt to search for the keybinding for the simple
```
Comment (carlos-zamora) [inline src/cascadia/TerminalApp/TerminalPage.cpp]:
```suggestion
                // First, attempt to search for the keybinding for the simple
```

File: src/cascadia/TerminalApp/AppKeyBindings.cpp (hunk 2 above)
Comment (zadjii-msft) [inline src/cascadia/TerminalApp/AppKeyBindings.cpp]:
```suggestion

    // Method Description:
```

**Comment by zadjii-msft** [comment]
I can't block my own change, but this needs to be merged nicely with #3629

**Pull request merged**